### PR TITLE
[SDK]automatic runtime improvements: allow "automatic" runtime name & add progress message

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR allows user specify "automatic" to use "automatic"
Added back progress message to keep user update with runtime creation progress.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/azure/operations/_run_operations.py [hunk 1]
```diff
@@ -682,20 +687,18 @@ def _resolve_automatic_runtime(self, run, session_id, reset=None):
...
 
-        if runtime:
-            if not isinstance(runtime, str):
-                raise TypeError(f"runtime should be a string, got {type(runtime)} for {runtime}")
-        else:
+        if runtime is None or runtime == AUTOMATIC_RUNTIME_NAME:
             runtime = self._resolve_automatic_runtime(run=run, session_id=session_id, reset=reset)
-
+        elif not isinstance(runtime, str):
+            raise TypeError(f"runtime should be a string, got {type(runtime)} for {runtime}")
```
Comment (0mza987) [inline src/promptflow/promptflow/azure/operations/_run_operations.py:701]: If no error raised, maybe worth a log to print what runtime is specified? Ignore this if it's printed in other place.

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py [hunk 2]
```diff
@@ -354,6 +354,13 @@ def submit(*args, **kwargs):
                 data=f"{DATAS_DIR}/env_var_names.jsonl",
             )
 
+        with patch.object(FlowServiceCaller, "submit_bulk_run") as mock_submit, patch.object(
+            RunOperations, "get"
+        ), patch.object(FlowServiceCaller, "create_flow_session"):
+            mock_submit.side_effect = submit
+            # automatic is a reserved runtime name, will use automatic runtime if specified.
+            pf.run(flow=f"{FLOWS_DIR}/print_env_var", data=f"{DATAS_DIR}/env_var_names.jsonl", runtime="automatic")
```
Comment (crazygao) [inline src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:362]: assert 2 times: both runtime = None and runtime = AUTOMATIC_RUNTIME_NAME should be covered.

Comment (D-W-) [reply src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py:362]: In both cases, the runtime_name will be "automatic" and it's covered in line 342

**Review by crazygao — APPROVED** [review]

File: src/promptflow/promptflow/azure/operations/_run_operations.py (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/azure/operations/_run_operations.py:701]: Offline synced, the runtime info already exists in flow run output dict.

**Pull request merged**